src/lib/shell/MainLoopDefault.cpp

```diff
@@ -31,71 +31,64 @@ using chip::Shell::streamer_get;
 
 namespace {
 
-ssize_t ReadLine(char * buffer, size_t max)
+// max > 1
+size_t ReadLine(char * buffer, size_t max)
 {
-    ssize_t read = 0;
-    bool done    = false;
-    char * inptr = buffer;
+    size_t line_sz = 0;
 
-    // Read in characters until we get a new line or we hit our max size.
-    while (((inptr - buffer) < static_cast<int>(max)) && !done)
+    // Read in characters until we get a line ending or EOT.
+    for (bool done = false; !done;)
```

Comment:
Suggest:
```
bool done = false;

while (!done) {
...
}

```

Reply:
I find the for-loop more clear and concise.

Reply:
Specifically, the for-loop explicitly shows that `done` is not used outside of the loop, which is information I would have appreciated at-a-glance when I was working on this.

If you think the style if more important than the semantics, please confirm and I will change it.